plugins/field-colour/src/blocks/generatorUtils.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type { Block } from 'blockly';
+import type { JavascriptGenerator} from 'blockly/javascript';
+import type { DartGenerator } from 'blockly/dart';
+import type { LuaGenerator } from 'blockly/lua';
+import type { PhpGenerator } from 'blockly/php';
+import type { PythonGenerator } from 'blockly/python';
+
+export type Generators = {
+    javascript?: JavascriptGenerator;
+    dart?: DartGenerator;
+    lua?: LuaGenerator;
+    php?: PhpGenerator;
+    python?: PythonGenerator;
+};
+
+export type JsBlockCodeGenerator = (
+    block: Block,
+    generator: JavascriptGenerator,
+) => [string, number] | string | null;
+
+
+export type DartBlockCodeGenerator = (
+    block: Block,
+    generator: DartGenerator,
+) => [string, number] | string | null;
```

Comment:
I _believe_ you can simplify this (and avoid type rot) by using [indexed access types](https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html):

```suggestion
export type DartBlockCodeGenerator = DartGenerator.forBlock[''];
```

Reply:
I get this error if I try that: "Exported type alias 'DartBlockCodeGenerator' has or is using private name 'DartGenerator'.ts(4081)"

Reply:
I am not sure what that means.  (Is it complaining that "`DartGenerator`" is a private name?  Huh?  What's with "`'DartGenerator'.ts`"??)

In any case if you want to export a type like `DartBlockCodeGenerator` I think it would be best if it were extracted from `DartGenerator` rather than constructed manually.

Reply:
Resolved by moving the block code generator installation into the per-block files, which gets rid of the need for a function signature and explicit type declarations. TS will still type-check appropriately.